### PR TITLE
Add some comments about how `quote_each_token` works.

### PR DESCRIPTION
**Pull request by nnethercote**
I didn't understand how this code worked until I used `trace_macros` on
an example and saw some expansions. This commit contains the comments I
wish had been present.

### CODE REVIEW

**Comment by nnethercote** [comment]
I updated the comment about performance -- the tt muncher is quadratic, this implementation is linear, but this implementation has higher constant factors and so is slower for short inputs. (I did some microbenchmarking to determine this.)

File: src/lib.rs [hunk 1]
```diff
@@ -669,6 +669,67 @@ macro_rules! quote_bind_next_or_break {
...
+// The middle tt (surrounded by parentheses) is the tt being processed.
+// - When it is a `#`, `quote_token_with_context!` can do an interpolation. The
+//   interpolation kind will depend on the three subsequent tts.
+// - When it is within a later part of an interpolation, it can be ignored,
+//   because the interpolation has already been done.
+// - When it is not part of an interpolation it can be treated normally.
+// - When it is a `@` it can be ignored.
+//
+// The parentheses around the middle tt aren't necessary, but they make the
+// macro definitions easier to read.
```
Comment (dtolnay) [inline src/lib.rs:732]: This part isn't right. I don't know how we would correctly drop the first 3 and last 3 quote_token_with_context calls without bracketing the middle token to make it unable to collide with the padding token (`@`).

I'll fix this in a followup PR.

**Review by dtolnay — APPROVED** [review]
Thanks! This is great.

File: src/lib.rs (hunk 1 above)
Comment (nnethercote) [reply src/lib.rs:732]: Ok, thanks!

**Pull request merged**